packages/sirius-web/frontend/sirius-web-application/src/views/edit-project/UndoRedo.tsx

```diff
@@ -0,0 +1,160 @@
+  payload.__typename === 'SuccessPayload';
+
+export const UndoRedo = ({ children }: { children: React.ReactNode }) => {
+  const [undo, { data: undoData }] = useMutation<GQLUndoData, GQLUndoVariables>(undoMutation);
+  const [redo, { data: redoData }] = useMutation<GQLRedoData, GQLUndoVariables>(redoMutation);
+  const { projectId } = useParams<EditProjectViewParams>();
+
+  useEffect(() => {
+    sessionStorage.setItem('undoStack', JSON.stringify([]));
+    sessionStorage.setItem('redoStack', JSON.stringify([]));
+  }, []);
+
+  const undoLastAction = () => {
+    var storedArray = sessionStorage.getItem('undoStack');
+    if (storedArray) {
+      var arr = JSON.parse(storedArray);
+      if (arr[0]) {
+        const input: GQLUndoInput = {
+          id: crypto.randomUUID(),
+          editingContextId: projectId,
+          mutationId: arr[0],
+        };
+        undo({ variables: { input } });
+      }
+    }
+  };
+
+  const redoLastAction = () => {
+    var storedArray = sessionStorage.getItem('redoStack');
+    if (storedArray) {
+      var arr = JSON.parse(storedArray);
+      if (arr[0]) {
+        const input: GQLRedoInput = {
+          id: crypto.randomUUID(),
+          editingContextId: projectId,
+          mutationId: arr[0],
+        };
+        redo({ variables: { input } });
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (undoData) {
+      const { undo } = undoData;
+      if (isSuccessPayload(undo)) {
+        var storedUndoStack = sessionStorage.getItem('undoStack');
+        var storedRedoStack = sessionStorage.getItem('redoStack');
+
+        //Remove first element of undo stack
+        var undoStack = JSON.parse(storedUndoStack);
+        var lastElement = undoStack.shift();
+        sessionStorage.setItem('undoStack', JSON.stringify(undoStack));
+
+        //Put the element in the 1st position of the redo stack
+        var redoStack = JSON.parse(storedRedoStack);
+        sessionStorage.setItem('redoStack', JSON.stringify([lastElement, ...redoStack]));
+      }
+    }
+  }, [undoData]);
+
+  useEffect(() => {
+    if (redoData) {
+      const { redo } = redoData;
+      if (isSuccessPayload(redo)) {
+        var storedUndoStack = sessionStorage.getItem('undoStack');
+        var storedRedoStack = sessionStorage.getItem('redoStack');
+
+        //Remove first element of redo stack
+        var redoStack = JSON.parse(storedRedoStack);
+        var lastElement = redoStack.shift();
+        sessionStorage.setItem('redoStack', JSON.stringify(redoStack));
+
+        //Put the element in the 1st position of the undo stack
+        var undoStack = JSON.parse(storedUndoStack);
+        sessionStorage.setItem('undoStack', JSON.stringify([lastElement, ...undoStack]));
+      }
+    }
+  }, [redoData]);
+
+  const undoKeyPressHandler = (e) => {
+    if (e.ctrlKey && e.key === 'z') {
+      undoLastAction();
+    }
+  };
+
+  const redoKeyPressHandler = (e) => {
+    if (e.ctrlKey && e.key === 'y') {
```

Comment:
Use `event.ctrlKey || event.metaKey` because it would not work on macOS otherwise. Ctrl key will be for windows / linux.